src/Swoole/Actions/EnsureRequestsDontExceedMaxExecutionTime.php

```diff
@@ -27,6 +27,10 @@ public function __invoke()
             if ((time() - $row['time']) > $this->maxExecutionTime) {
                 $this->timerTable->del($workerId);
 
+                if ($this->server instanceof Server && ! $this->server->exists($row['fd'])) {
```

Comment:
Looks like a good patch. But what is the reproduction code? 

When I test with this code, I close the browser and `exists` still return `true`.

```php
$http = new Swoole\Http\Server('0.0.0.0', 9502);

$http->on('Request', function ($request, $response) use ($http) {
    sleep(10);
    var_dump($http->exists($request->fd));
    $response->end('Hello Swoole');
});

$http->start();
```

Can you add a test case?

Reply:
I encountered this issue only in a production environment and within a containerized setup. It seems that others who reported the issue are facing the same situation. However, despite my attempts, I was unable to reproduce this issue in a test case.